src/envoy/http/service_control/filter_stats.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/protobuf/stubs/status.h"
+
+#include "src/envoy/http/service_control/filter_stats.h"
+
+using ::google::protobuf::util::Status;
+using ::google::protobuf::util::error::Code;
+
+namespace espv2 {
+namespace envoy {
+namespace http_filters {
+namespace service_control {
+
+void ServiceControlFilterStats::collectCheckStatus(
```

Comment:
Can we find a less redundant way of doing this? Templating would certainly help

How does Envoy do this? They collect stats per http code, right?

Reply:
I don't see how templating can help cut down line of code. Could you give an example if you don't mind?

As I see in envoy's repo, they don't have the case of collecting all the status.